platform/haiku/detect.py

```diff
@@ -36,16 +36,21 @@ def configure(env):
     ## Build type
 
     if (env["target"] == "release"):
-        if (env["debug_release"] == "yes"):
+        env.Prepend(CCFLAGS=['-O3', '-ffast-math'])
+        if (env["debug_symbols"] == "yes"):
+            env.Prepend(CCFLAGS=['-g1'])
+        if (env["debug_symbols"] == "full"):
             env.Prepend(CCFLAGS=['-g2'])
-        else:
-            env.Prepend(CCFLAGS=['-O3', '-ffast-math'])
 
     elif (env["target"] == "release_debug"):
         env.Prepend(CCFLAGS=['-O2', '-ffast-math', '-DDEBUG_ENABLED'])
+        if (env["debug_symbols"] == "yes"):
+            env.Prepend(CCFLAGS=['-g1'])
+        if (env["debug_symbols"] == "full"):
+            env.Prepend(CCFLAGS=['-g2'])
 
     elif (env["target"] == "debug"):
-        env.Prepend(CCFLAGS=['-g2', '-DDEBUG_ENABLED', '-DDEBUG_MEMORY_ENABLED'])
+        env.Prepend(CCFLAGS=['-g3', '-DDEBUG_ENABLED', '-DDEBUG_MEMORY_ENABLED'])
```

Comment:
Do we really gain a lot of information with this, or just a lot of wasted disk space? :P

Reply:
Yeah, specifically we gain macro expansions in the debug information meaning you can set breakpoints in and on macros, and can step through them. Given that most of the extra debuggable information in the debug build comes from memory checking macros this seemed like a useful thing to have.

The difference between -g1 and -g2 is much larger than the difference between -g2 and -g3.